### ISSUE TITLE
Allow running a shell command like shell_gpt

### ISSUE THREAD

**Issue by maelp** [charmbracelet/mods#152]
https://github.com/TheR1D/shell_gpt has a nice `sgpt -s 'print me the size of all the files in the src/ dir'` which then offers to execute, describe (or cancel) the command returned by chat-gpt, it's very convenients for one-off (and caches the results to rerun them multiple times without hitting the API)

also nice with `--editor` to create the prompt in an editor

similarly https://github.com/KillianLucas/open-interpreter allows running arbitrary code, and could be a nice integration with the beautiful charm UI

**Comment by jzyinq** [comment]
As for `--editor` part you can mimic this flag yourself with such oneliner:
```bash
tempfile="$(mktemp)" && $EDITOR "$tempfile" && cat "$tempfile" | mods -P
```
Just put it in some memorable alias and you're good to go.

**Comment by caarlos0** [comment]
if you run mods with no args, it'll open an editor (since a couple of versions ago)

**Comment by jzyinq** [comment]
> if you run mods with no args, it'll open an editor (since a couple of versions ago)

That's true - I'm just not a fan of `alt-enter` as a line breaker so it's easier for me to use vim anyway.

**Comment by jzyinq** [comment]
Nice, I didn't know this.

**Comment by caarlos0** [comment]
this will probably require functions/tools #212 

i think we can close this and leave #212 only?
